From 1109c847b272be6b65ec3d2a4c58e91f0be369fc Mon Sep 17 00:00:00 2001
From: Matthias Klumpp <matthias@tenstral.net>
Date: Thu, 3 Sep 2020 00:07:48 +0200
Subject: [PATCH] validator: Don't pass NULL to vprintf, it's not an allowed
 value

The format parameter is listed as non-nullable in the API documentation,
and crashes were reported indicating that this does cause real-world
issues. This likely fixes
https://github.com/ximion/appstream-generator/issues/83
---
 src/as-validator.c | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/src/as-validator.c b/src/as-validator.c
index 32704b70..53d87f8f 100644
--- a/src/as-validator.c
+++ b/src/as-validator.c
@@ -158,9 +158,11 @@ as_validator_add_issue (AsValidator *validator, xmlNode *node, const gchar *tag,
 		explanation = tag_data->explanation;
 	}
 
-	va_start (args, format);
-	buffer = g_strdup_vprintf (format, args);
-	va_end (args);
+	if (format != NULL) {
+		va_start (args, format);
+		buffer = g_strdup_vprintf (format, args);
+		va_end (args);
+	}
 
 	issue = as_validator_issue_new ();
 	as_validator_issue_set_tag (issue, tag_final);
